cmd/rebasehelpers/util/git.go

```diff
@@ -0,0 +1,97 @@
+package util
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+type Commit struct {
+	Sha     string
+	Summary string
+	Files   []File
+}
+
+func (c Commit) DeclaresUpstreamChange() bool {
+	return strings.Contains(strings.ToLower(c.Summary), "upstream")
```

Comment:
If you had such a summary with no changes to `Godeps/...` then you'd get dinged with an error. A consequence of the validation strategy overall is that "upstream" becomes a reserved word in summaries, otherwise we can't validate two-ways (undeclared with godeps changes, declared without godeps changes).